environment/k8s_environment.go

```diff
@@ -169,6 +177,69 @@ func (env *k8sEnvironment) TearDown() {
 	log.Error().Err(err)
 }
 
+// On a test failure, dump all the pod logs
+func (env *k8sEnvironment) logFailures() {
+	// Create logs directory if non-existent
+	log.Info().Msg("Creating log files on test failure")
+	logsFolder := tools.ProjectRoot + "/logs/"
+	if _, err := os.Stat(logsFolder); os.IsNotExist(err) {
+		if err = os.Mkdir(logsFolder, 0755); err != nil {
+			log.Err(err).Str("Folder Name", logsFolder).Msg("Error creating logs directory")
+		}
+	}
+
+	// Create specific test folder
+	testLogFolder := logsFolder + strings.Replace(ginkgo.CurrentGinkgoTestDescription().TestText, " ", "-", -1) +
+		"_" + env.namespace.Name + "/"
+	if _, err := os.Stat(testLogFolder); os.IsNotExist(err) {
+		if err = os.Mkdir(testLogFolder, 0755); err != nil {
+			log.Err(err).Str("Folder Name", testLogFolder).Msg("Error creating logs directory")
+		}
+	}
+
+	// Get logs from K8s pods
+	podsClient := env.k8sClient.CoreV1().Pods(env.namespace.Name)
+	podsList, err := podsClient.List(context.Background(), metaV1.ListOptions{})
+	if err != nil {
+		log.Err(err).Str("Env Name", env.namespace.Name).Msg("Error retrieving pod list from K8s environment")
+	}
+
+	// Each Pod gets a file
+	for _, pod := range podsList.Items {
+		podName := pod.Labels[SelectorLabelKey]
+		logFile, err := os.Create(filepath.Join(testLogFolder, podName) + ".log")
+		if err != nil {
+			log.Err(err).Str("File Name", logFile.Name()).Msg("Error creating log file")
+		}
+
+		_, _ = logFile.WriteString("======================================================================================\n")
+		_, _ = logFile.WriteString("Test file name: " + ginkgo.CurrentGinkgoTestDescription().FileName + "\n")
+		_, _ = logFile.WriteString("Test file line: " + fmt.Sprint(ginkgo.CurrentGinkgoTestDescription().LineNumber) + "\n")
+		_, _ = logFile.WriteString("======================================================================================\n")
+
+		podLogOptions := &coreV1.PodLogOptions{}
+		if strings.Contains(podName, "chainlink") {
+			podLogOptions = &coreV1.PodLogOptions{Container: "node"}
+		}
+
+		podLogRequest := podsClient.GetLogs(pod.Name, podLogOptions)
+		podLogs, err := podLogRequest.Stream(context.Background())
+		if err != nil {
+			log.Err(err).Str("Env Name", env.namespace.Name).Msg("error in opening pod logging stream")
+		}
+
+		buf := new(bytes.Buffer)
+		_, err = io.Copy(buf, podLogs)
+		if err != nil {
+			log.Err(err).Str("Env Name", env.namespace.Name).Msg("error in copying information from podLogs to a buffer")
```

Comment:
k8s specific, but logging has been consistent in using `Namespace` for the key in the log output.